confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/input/ConfluenceXMLPackage.java

```diff
@@ -1816,6 +1816,11 @@ private String fixCDataAndNL(String elementText)
             .replace('\u2029', '\n');
     }
 
+    private String escapeWindowsFolderName(String folderName)
+    {
+        return folderName.replace("[<>:\"/\\|?*]", "");
```

Comment:
Hmm, removing characters which cause problems is a risk of collisions. As suggested in a previous comment, the safest would be to use some encoding (for example URL encoding). You might want to take a look at https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-store/xwiki-platform-store-filesystem/src/main/java/org/xwiki/store/internal/FileSystemStoreUtils.java#L59, for inspiration as it's internal (but maybe commons-io or commons-lang have some tools like this).

Reply:
Found this class of commons.io that has this method https://github.com/apache/commons-io/blob/master/src/main/java/org/apache/commons/io/FileSystem.java#L512 . It basically replaces the illegal characters of a given filesystem with a given character. The risk of collisions would still be there but i don't think in our case we need to worry about that too hard. As of now, this type of id `557058:403042db-7e36-4f27-8739-b3f5fef3d15d` is the only one that we had problems with. For this example, we would encounter a collision if there would exist an id that matches this regex `557058[<>"/\|?*]403042db-7e36-4f27-8739-b3f5fef3d15d`. WDYT? Should I use this or stick with the URL encoding

Reply:
At least the good thing is that you would have a behavior which depend on the current system with this tool, so it would reduce the risk of collision to mainly Windows. Replacing by a character is also safer than a remove, even if not perfect. I guess we can start with that (it's better than crashing anyway) and see if we ever have a collision problem...